source/common/common/trie.h

```diff
@@ -0,0 +1,111 @@
+#pragma once
+
+#include <memory>
```

Comment:
nit: del standard includes like these and add to precompiled.h if not already there. Same applies elsewhere.